src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java

```diff
@@ -343,6 +344,14 @@ public CcCompilationHelper fromCommon(CcCommon common, ImmutableList<String> add
     return this;
   }
 
+  /** Sets fields that only exist in cc_library rules. */
+  public CcCompilationHelper fromLibrary(RuleContext ruleContext) {
```

Comment:
For public methods of CcCompilationHelper, please do not pass RuleContext attributes. The public methods of CcCompilationHelper are essentially the API of cc_common.compile() and we cannot accept RuleContext there. Just leave the public method here that takes a list of CcCompilationContexts.